File: frontend/src/types/knowledge.ts
export interface Knowledge {
  id: string
  title: string
  content?: string
  knowledge_type: string
  file_url?: string
  file_name?: string
  tags: string[]
  category?: string
  reference_count: number
  created_at: string
  updated_at: string
}

export interface KnowledgeCreate {
  title: string
  content?: string
  type?: 'text' | 'file'
  tags?: string[]
  category?: string
}

export interface KnowledgeUpdate {
  title?: string
  content?: string
  tags?: string[]
  category?: string
}

File: backend/app/models/knowledge.py
from datetime import datetime
from typing import List, Literal, Optional

from pydantic import BaseModel


class KnowledgeBase(BaseModel):
    title: str
    content: Optional[str] = None
    type: Literal["text", "file"] = "text"
    tags: List[str] = []
    category: Optional[str] = None


class KnowledgeCreate(KnowledgeBase):
    pass


class KnowledgeUpdate(BaseModel):
    title: Optional[str] = None
    content: Optional[str] = None
    tags: Optional[List[str]] = None
    category: Optional[str] = None


class Knowledge(KnowledgeBase):
    id: str
    file_url: Optional[str] = None
    file_name: Optional[str] = None
    reference_count: int = 0
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True
